### PR TITLE
#10 Task. Sergey Starushkin

### PR DESCRIPTION
**Pull request by Sergey-Starushkin**

### CODE REVIEW

File: task/10-katas-1-tasks.js [hunk 1]
```diff
@@ -56,7 +88,27 @@ function createCompassPoints() {
...
+        regex = new RegExp("\{([0-9a-zA-Z\.,]+)\}", 'i'),
+        itemSet = new Set;
+
+    queue.push(str);
+
+    while (queue.length) {
+        let item = queue.shift();
+        let matches = item.match(regex);
+
+        if (matches != null) {
```
Comment (vramaniuk) [inline task/10-katas-1-tasks.js:101]: "operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"

File: task/10-katas-1-tasks.js [hunk 2]
```diff
@@ -113,7 +186,33 @@ function getZigZagMatrix(n) {
  *
  */
 function canDominoesMakeRow(dominoes) {
-    throw new Error('Not implemented');
+    function task10(current, value, left) {
+        if (left == 0) {
```
Comment (vramaniuk) [inline task/10-katas-1-tasks.js]: operator === is always preferable
sometimes ''==' may cause an error
(check at console :
'0' == 0
and '0' === 0)

File: task/10-katas-1-tasks.js [hunk 3]
```diff
@@ -113,7 +186,33 @@ function getZigZagMatrix(n) {
...
+            result = true;
+            return;
+        }
+
+        visited[current] = true;
+
+        for (let i = 0; i < dominoes.length; i++)
+            if (!visited[i]) {
+                if (dominoes[i].indexOf(value) != -1) {
```
Comment (vramaniuk) [inline task/10-katas-1-tasks.js]: operator !== is always preferable

File: task/10-katas-1-tasks.js [hunk 4]
```diff
@@ -113,7 +186,33 @@ function getZigZagMatrix(n) {
...
+                    task10(i, dominoes[i][0] == value ? dominoes[i][1] : dominoes[i][0], left - 1);
```
Comment (vramaniuk) [inline task/10-katas-1-tasks.js]: operator === is always preferable

File: task/10-katas-1-tasks.js [hunk 5]
```diff
@@ -88,7 +140,28 @@ function* expandBraces(str) {
...
+    });
+    function overallCode(v1, v2) {
+        if (v1 < n) v1++;else v2 += 2;
+        if (v2 > 1) v2--;
+    }
+    var i = 1,
+        j = 1;
+    for (var k = 0; k < n * n; k++) {
+        arr[i - 1][j - 1] = k;
+        if ((i + j) % 2 == 0) {
```
Comment (vramaniuk) [inline task/10-katas-1-tasks.js]: operator === is always preferable

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by vramaniuk** [comment]
Please, privide the direct link to last successful travis build

**Comment by Sergey-Starushkin** [comment]
https://travis-ci.org/github/Sergey-Starushkin/js-assignments/builds/712295731

**Comment by vramaniuk** [comment]
Dirty PR. Please, see notes for tasks 1-4

And implement previus notes (strict equality)

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Pull request closed without merging**